### PR TITLE
CLOUDSTACK-6900: This fixes volume migration failing after that volume is downloaded

### PR DESCRIPTION
**Pull request by anshul1886**
When we download volume then we create entry in volume_store_ref table.
We mark the volume entry to ready state once download_url gets generated.
When we migrate that volume, then again one more entry is created with same volume id.
Its state is marked as allocated. Later we try to list only one dataobject in datastore
for state transition during volume migration. If the listed volume's state is allocated
then migration passes otherwise it fails.

Below fix will remove the randomness and give priority to volume entry which is made for
migration (download_url/extracturl will be null in case of migration). Giving priority to
download volume case is not needed as there will be only one entry in that case so no randomness.

### CODE REVIEW

**Comment by bhaisaab** [comment]
@anshul1886 Good workaround, though relying on null state is probably not a good idea? Is there a way you can detect that without using null-ness but a state associated with the volume entry?

**Comment by anshul1886** [comment]
If not download_url than there needs to be other entry in db to detect the difference type of entries. But that will require the DB change and will not achieve any extra purpose.

I didn't get second part of question as these entries are made basically to track the state of volume.

**Comment by bhaisaab** [comment]
Let me rephrase - are you certain that the extract url (download url) of the volume that needs migration will be always null? Consider possible schema changes. I was thinking to explicitly have a field (instead of download_url being null).

**Comment by anshul1886** [comment]
Yes it will always be null for migration entries.

**Comment by bhaisaab** [comment]
Comment, should we merge it? TravisCI is complaining?

**Comment by anshul1886** [comment]
Yes. It seems travis CI errored out before running any tests.
